src/components/CatalogSearchCriteria.tsx

```diff
@@ -19,130 +19,216 @@ const DATE_FORMAT = 'YYYY-MM-DD'
+                onChange={this.handleDateOfCaptureFromChange}
+              />
+            </label>
+            {isValidDate(this.props.catalog.searchCriteria.dateFrom) || <div className={styles.invalidDates}>
+              Invalid date ({date_format})
+            </div>}
+            <label className={styles.captureDateTo}>
+              <span>To</span>
+              <input
+                value={this.props.catalog.searchCriteria.dateTo}
+                type="text"
+                pattern="[0-9]{4}-[0-9]{2}-[0-9]{2}"
+                placeholder={date_format}
+                title={date_format}
+                disabled={this.props.catalog.isSearching}
+                onChange={this.handleDateOfCaptureToChange}
+              />
+            </label>
+            {isValidDate(this.props.catalog.searchCriteria.dateTo) || <div className={styles.invalidDates}>
+              Invalid date ({date_format})
+            </div>}
+            {isValidDateRange(this.props.catalog.searchCriteria.dateFrom, this.props.catalog.searchCriteria.dateTo) || <div className={styles.invalidDates}>
+              From date must be before To date.
+            </div>}
+          </div>
+        )}
+
+        <h3>Filters</h3>
+        <label className={styles.cloudCover}>
+          <span>Cloud Cover</span>
           <input
-            value={props.dateTo}
-            type="text"
-            pattern="[0-9]{4}-[0-9]{2}-[0-9]{2}"
-            placeholder={date_format}
-            title={date_format}
-            disabled={props.disabled}
-            onChange={event => props.onDateChange(
-              props.dateFrom,
-              (event.target as HTMLInputElement).value
-            )}
+            value={this.props.catalog.searchCriteria.cloudCover.toString()}
+            type="range"
+            min="0"
+            max="100"
+            onChange={this.handleCloudCoverChange}
           />
-        </label>
-        {isValidDate(props.dateTo) || <div className={styles.invalidDates}>
-          Invalid date ({date_format})
-        </div>}
-        {isValidDateRange(props.dateFrom, props.dateTo) || <div className={styles.invalidDates}>
-          From date must be before To date.
-        </div>}
-      </div>
-    )}
-
-    <h3>Filters</h3>
-    <label className={styles.cloudCover}>
-      <span>Cloud Cover</span>
-      <input
-        value={props.cloudCover.toString()}
-        type="range"
-        min="0"
-        max="100"
-        onChange={event => props.onCloudCoverChange(
-          parseInt((event.target as HTMLInputElement).value, 10)
-        )}
-      />
-      <span className={styles.value}>
+          <span className={styles.value}>
         <span className={styles.leSign}>&le;&thinsp;</span>
-        {props.cloudCover}%
+            {this.props.catalog.searchCriteria.cloudCover}%
       </span>
-    </label>
-  </div>
-)
+        </label>
+      </div>
+    )
+  }
+
+  private handleSourceChange(e: FormEvent) {
```

Comment:
I think instead of casting e.target you _might_ be able to do`e: FormEvent<HTMLSelectElement>`

Reply:
I'm getting a compile error when I try doing that. It does seem like there should be a better way than casting it, but I guess I'll just leave it for now.